gpflow/densities.py

```diff
@@ -77,11 +77,11 @@ def multivariate_normal(x, mu, L):
     assume independence over the *columns*: the number of rows must match the
     size of L.
     """
-    d = x - mu
+    d = tf.cond(tf.rank(x) < 2, lambda: x[:, None], lambda: x) \
```

Comment:
Add some readability to the code:
```
x = tf.cond(tf.rank(x) < 2, lambda: x[:, None], lambda: x)
mu = tf.cond(tf.rank(mu) < 2, lambda: mu[:, None], lambda: mu)
d = x - mu
```
and num_col and num_dims types depend on L type
```
num_col = tf.cast(tf.shape(d)[1], L.dtype)
num_dims = tf.cast(tf.shape(d)[0], L.dtype)
```

Reply:
Sounds good! (the use of settings.float_type was already there but your change sounds sensible)